cmd/gctcli/futures.go

```diff
@@ -298,6 +217,268 @@ var futuresCommands = &cli.Command{
+					Name:    "exchange",
+					Aliases: []string{"e"},
+					Usage:   "the exchange to retrieve futures positions from",
+				},
+				&cli.StringFlag{
+					Name:    "asset",
+					Aliases: []string{"a"},
+					Usage:   "the asset type of the currency pair, must be a futures type",
+				},
+				&cli.StringFlag{
+					Name:    "pair",
+					Aliases: []string{"p"},
+					Usage:   "the currency pair",
+				},
+				&cli.StringFlag{
+					Name:    "margintype",
+					Aliases: []string{"margin", "mt", "m"},
+					Usage:   "the margin type, such as 'isolated', 'multi' or 'cross'",
+				},
+			},
+		},
+		{
+			Name:      "changepositionmargin",
+			Aliases:   []string{"cpm"},
+			Usage:     "sets isolated margin levels for an existing position",
+			ArgsUsage: "<exchange> <asset> <pair> <start>",
+			Action:    changePositionMargin,
+			Flags: []cli.Flag{
+				&cli.StringFlag{
+					Name:    "exchange",
+					Aliases: []string{"e"},
+					Usage:   "the exchange to retrieve futures positions from",
+				},
+				&cli.StringFlag{
+					Name:    "asset",
+					Aliases: []string{"a"},
+					Usage:   "the asset type of the currency pair, must be a futures type",
+				},
+				&cli.StringFlag{
+					Name:    "pair",
+					Aliases: []string{"p"},
+					Usage:   "the currency pair",
+				},
+				&cli.StringFlag{
+					Name:    "margintype",
+					Aliases: []string{"margin", "mt", "m"},
+					Usage:   "the margin type, most likely 'isolated'",
+				},
+				&cli.Float64Flag{
+					Name:    "originalallocatedmargin",
+					Aliases: []string{"oac"},
+					Usage:   "the original allocated margin, is used by some exchanges to determine differences to apply",
+				},
+				&cli.Float64Flag{
+					Name:    "newallocatedmargin",
+					Aliases: []string{"nac"},
+					Usage:   "the new allocated margin level you desire",
+				},
+				&cli.StringFlag{
+					Name:    "marginside",
+					Aliases: []string{"side", "ms"},
+					Usage:   "optional - the margin side, typically 'buy' or 'sell'",
+				},
+			},
+		},
+		{
+			Name:      "getfuturespositionsummary",
+			Aliases:   []string{"summary", "fps"},
+			Usage:     "return a summary of your futures position",
+			ArgsUsage: "<exchange> <asset> <pair> <start>",
+			Action:    getFuturesPositionSummary,
+			Flags: []cli.Flag{
+				&cli.StringFlag{
+					Name:    "exchange",
+					Aliases: []string{"e"},
+					Usage:   "the exchange to retrieve futures positions from",
+				},
+				&cli.StringFlag{
+					Name:    "asset",
+					Aliases: []string{"a"},
+					Usage:   "the asset type of the currency pair, must be a futures type",
+				},
+				&cli.StringFlag{
+					Name:    "pair",
+					Aliases: []string{"p"},
+					Usage:   "the currency pair",
+				},
+				&cli.StringFlag{
+					Name:    "underlyingpair",
+					Aliases: []string{"up"},
+					Usage:   "optional - the underlying currency pair eg if pair is BTCUSD-1984-C, the underlying pair could be BTC-USD",
+				},
+			},
+		},
+		{
+			Name:      "getfuturepositionorders",
```

Comment:
Returns an empty position. 
```
go run ./... futures getfuturepositionorders --exchange=okx --asset=futures -pair=BTC-USDC-230929
{
 "positions": [
  {
   "exchange": "Okx",
   "asset": "futures",
   "pair": {
    "delimiter": "-",
    "base": "BTC",
    "quote": "USDC-230929"
   }
  }
 ]
}
```

Reply:
Do you have any positions?

Reply:
nope, should have mentioned that sorry.

Reply:
Are you saying you would like a len check and return an error instead? It will append any orders in the response if they exist

Reply:
I was expecting nothing back to be honest. Just `{}` If its not context feasible then this is pretty minor you can skip.